test/test_jit.py

```diff
@@ -3984,6 +3984,97 @@ def test_while(a, b):
                 return a + b
             ''')
 
+    def test_optional_refinement(self):
+        @torch.jit.script
+        def test_if_none_assignment(x):
+            # type: (Optional[int]) -> int
+            if x is None:
+                x = 1
+            return x + 1
+
+        @torch.jit.script
+        def test_ternary(x):
+            # type: (Optional[int]) -> int
```

Comment:
It looks like this doesn't do any refinement (both branches return a literal)

Reply:
Good catch